clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java

```diff
@@ -247,7 +249,31 @@ public EhcacheEntityResponse invoke(ClientDescriptor clientDescriptor, EhcacheEn
 
   @Override
   public void handleReconnect(ClientDescriptor clientDescriptor, byte[] extendedReconnectData) {
-    //nothing to do
+    ClientState clientState = this.clientStateMap.get(clientDescriptor);
+    if (clientState == null) {
+      LOGGER.error("Client {} trying to reconnect is not connected to entity.", clientDescriptor);
+      return;
+    }
+    clientState.attach();
+    Set<String> cacheIds = reconnectDataCodec.decode(extendedReconnectData);
+    for (final String cacheId : cacheIds) {
+      ServerStoreImpl serverStore = ehcacheStateService.getStore(cacheId);
+      if (serverStore == null) {
+        //Client removes the cache's reference only when destroy has successfully completed
+        //This happens only when client thinks destroy is still not complete
```

Comment:
Got precision from @AbfrmBlr - given that, the following log statement should be `warn` level.